NuRadioReco/modules/io/rno_g/readRNOGDataMattak.py

```diff
@@ -0,0 +1,644 @@
+      
+      select: bool
+         Return True to select an dataset, return False to reject/skip it.
+      """
+     
+      # get first eventInfo
+      dataset.setEntries(0)
+      event_info = dataset.eventInfo()
+      
+      run_id = event_info.run
+      station_id = event_info.station
+      
+      run_info = self.__run_table.query(f"station == {station_id:d} & run == {run_id:d}")
+      
+      # "time_start/end" is stored in the isot format. datetime is much faster than astropy (~85ns vs 55 mus).
+      # But using datetime would mean to stip decimals because datetime can only handle mu sec precision and can not cope 
+      # with the additional decimals for ns.
+      if self._time_low is not None:
+         time_end = astropy.time.Time(run_info["time_end"].values[0])
+         if time_end < self._time_low:
+            self.logger.info(f"Reject station {station_id} run {run_id} because run ended before {self._time_low}")
+            return False
+      
+      if self._time_high is not None:
+         time_start = astropy.time.Time(run_info["time_start"].values[0])
+         if time_start > self._time_high:
+            self.logger.info(f"Reject station {station_id} run {run_id} because run started time after {self._time_high}")
+            return False
+      
+      run_type = run_info["run_type"].values[0]
+      if not run_type in self.__run_types:
+         self.logger.info(f"Reject station {station_id} run {run_id} because of run type {run_type}")
+         return False
+      
+      trigger_rate = run_info["trigger_rate"].values[0] * units.Hz 
+      if self.__max_trigger_rate and trigger_rate > self.__max_trigger_rate:
+         self.logger.info(f"Reject station {station_id} run {run_id} because trigger rate is to high ({trigger_rate / units.Hz} Hz)")
+         return False
+      
+      return True
+
+
+   def __get_n_events_of_prev_datasets(self, dataset_idx):
+      """ Get accumulated number of events from previous datasets """
+      dataset_idx_prev = dataset_idx - 1
+      return int(self._event_idxs_datasets[dataset_idx_prev]) if dataset_idx_prev >= 0 else 0
+      
+   
+   def __get_dataset_for_event(self, event_idx):
+      """ Get correct dataset and set entry accordingly to event index
+      
+      Parameters
+      ----------
+      
+      event_index: int
+         Same as in read_event().
+      
+      Returns
+      -------
+      
+      dataset: mattak.Dataset.Dataset
+      """
+      # find correct dataset
+      dataset_idx = np.digitize(event_idx, self._event_idxs_datasets)
+      dataset = self._datasets[dataset_idx]
+
+      event_idx_in_dataset = event_idx - self.__get_n_events_of_prev_datasets(dataset_idx)
+      dataset.setEntries(event_idx_in_dataset)  # increment iterator -> point to new event
+            
+      return dataset
+   
+   
+   def filter_event(self, evtinfo, event_idx=None):
+      """ Filter an event base on its EventInfo and the configured selectors.
```

Comment:
Should this also be a private method? I'm again not quite sure about its use case. I think you don't currently expose any `mattak.Dataset.EventInfo` anywhere, in any case.

Reply:
ok